spec/router/dsl_spec.rb

```diff
@@ -186,4 +186,39 @@
       end
     end
   end
+
+  context "with the match helper" do
+    it "correctly adds handlers" do
+      expect(router).to receive(:on).with("GET", "/test", "test#index", constraints: {}, defaults: nil)
+      expect(router).to receive(:on).with("POST", "/test", "test#index", constraints: {}, defaults: nil)
+
+      dsl.draw do
+        match "/test", to: "test#index", via: [:get, :post]
+      end
+    end
+
+    it "correctly adds handlers on via: :all" do
+      expect(router).to receive(:on).with("GET", "/test", "test#index", constraints: {}, defaults: nil)
+      expect(router).to receive(:on).with("POST", "/test", "test#index", constraints: {}, defaults: nil)
+      expect(router).to receive(:on).with("PUT", "/test", "test#index", constraints: {}, defaults: nil)
+      expect(router).to receive(:on).with("PATCH", "/test", "test#index", constraints: {}, defaults: nil)
+      expect(router).to receive(:on).with("DELETE", "/test", "test#index", constraints: {}, defaults: nil)
+
+      dsl.draw do
+        match "/test", to: "test#index", via: :all
+      end
+    end
+
+    it "correctly routes to all when no via is specified" do
+      expect(router).to receive(:on).with("GET", "/test", "test#index", constraints: {}, defaults: nil)
+      expect(router).to receive(:on).with("POST", "/test", "test#index", constraints: {}, defaults: nil)
+      expect(router).to receive(:on).with("PUT", "/test", "test#index", constraints: {}, defaults: nil)
+      expect(router).to receive(:on).with("PATCH", "/test", "test#index", constraints: {}, defaults: nil)
+      expect(router).to receive(:on).with("DELETE", "/test", "test#index", constraints: {}, defaults: nil)
+
+      dsl.draw do
+        match "/test", to: "test#index"
+      end
+    end
+  end
```

Comment:
I love the tests!

Can you please also add one to check if constraints and defaults are correctly passed down? And another one to check if `match` works correctly inside `scope`.